persist/fs/msgpack/schema.go

```diff
@@ -49,37 +74,75 @@ const (
+	minNumIndexSummaryFields         = 3
+	minNumLogInfoFields              = 3
+	minNumLogEntryFields             = 7
+	minNumLogMetadataFields          = 3
+
+	// curr number of fields specifies the number of fields that the current
+	// version of the M3DB will encode. This is used to ensure that the
+	// correct number of fields is encoded into the files. These values need
+	// to be incremened whenever we add new fields to an object.
+	currNumRootObjectFields           = 2
+	currNumIndexInfoFields            = 8
+	currNumIndexSummariesInfoFields   = 1
+	currNumIndexBloomFilterInfoFields = 2
+	currNumIndexEntryFields           = 5
+	currNumIndexSummaryFields         = 3
+	currNumLogInfoFields              = 3
+	currNumLogEntryFields             = 7
+	currNumLogMetadataFields          = 3
 )
 
-var numObjectFields []int
+var minNumObjectFields []int
+var currNumObjectFields []int
 
-func numFieldsForType(objType objectType) int {
-	return numObjectFields[int(objType)-1]
+func numFieldsForType(objType objectType) (min, curr int) {
+	return minNumObjectFields[int(objType)-1], currNumObjectFields[int(objType)-1]
 }
 
-func setNumFieldsForType(objType objectType, numFields int) {
-	numObjectFields[int(objType)-1] = numFields
+func setMinNumObjectFieldsForType(objType objectType, numFields int) {
+	minNumObjectFields[int(objType)-1] = numFields
+}
+
+func setCurrNumObjectFieldsForType(objType objectType, numFields int) {
+	currNumObjectFields[int(objType)-1] = numFields
 }
 
 func init() {
-	numObjectFields = make([]int, int(numObjectTypes))
+	minNumObjectFields = make([]int, int(numObjectTypes))
+	currNumObjectFields = make([]int, int(numObjectTypes))
+
+	setMinNumObjectFieldsForType(rootObjectType, minNumRootObjectFields)
+	setMinNumObjectFieldsForType(indexInfoType, minNumIndexInfoFields)
+	setMinNumObjectFieldsForType(indexSummariesInfoType, minNumIndexSummariesInfoFields)
+	setMinNumObjectFieldsForType(indexBloomFilterInfoType, minNumIndexBloomFilterInfoFields)
+	setMinNumObjectFieldsForType(indexEntryType, minNumIndexEntryFields)
+	setMinNumObjectFieldsForType(indexSummaryType, minNumIndexSummaryFields)
+	setMinNumObjectFieldsForType(logInfoType, minNumLogInfoFields)
+	setMinNumObjectFieldsForType(logEntryType, minNumLogEntryFields)
+	setMinNumObjectFieldsForType(logMetadataType, minNumLogMetadataFields)
 
-	setNumFieldsForType(rootObjectType, numRootObjectFields)
-	setNumFieldsForType(indexInfoType, numIndexInfoFields)
-	setNumFieldsForType(indexSummariesInfoType, numIndexSummariesInfoFields)
-	setNumFieldsForType(indexBloomFilterInfoType, numIndexBloomFilterInfoFields)
-	setNumFieldsForType(indexEntryType, numIndexEntryFields)
-	setNumFieldsForType(indexSummaryType, numIndexSummaryFields)
-	setNumFieldsForType(logInfoType, numLogInfoFields)
-	setNumFieldsForType(logEntryType, numLogEntryFields)
-	setNumFieldsForType(logMetadataType, numLogMetadataFields)
+	setCurrNumObjectFieldsForType(rootObjectType, currNumRootObjectFields)
```

Comment:
how about adding an assertion here:

```
func mustBeGreaterThanEqual(x, y int) {
 if x < y {
    panic(fmt.Sprintf("expected %d to be greater than or equal to %d", x, y))
 }
}

func init() {
  ...

  mustBeGreaterThanEqual(currNumRootObjectFields, minNumRootObjectFields)
  setCurrNumObjectFieldsForType(rootObjectType, currNumRootObjectFields)
  // same for other curr{fieldName} ...
}
```

Reply:
Hm yeah, perhaps before all the setting calls though, don't want it to get too messy here.